src/auth/ha-auth-external.ts

```diff
@@ -0,0 +1,65 @@
+import type { CSSResultGroup, TemplateResult } from "lit";
+import { css, html } from "lit";
+import { customElement, property } from "lit/decorators";
+import type { DataEntryFlowStepExternal } from "../data/data_entry_flow";
+import { fireEvent } from "../common/dom/fire_event";
+import type { LocalizeFunc } from "../common/translations/localize";
+import { HaForm } from "../components/ha-form/ha-form";
+
+@customElement("ha-auth-external")
+export class HaAuthExternal extends HaForm {
+  @property({ attribute: false }) public localize!: LocalizeFunc;
+
+  @property({ attribute: false }) public stepTitle?: string;
+
+  @property({ attribute: false }) public step!: DataEntryFlowStepExternal;
+
+  protected render(): TemplateResult {
+    return html`
+      <h2>${this.stepTitle}</h2>
+      <div class="content">
+        ${this.localize("ui.panel.page-authorize.external.description")}
+        <div class="open-button">
+          <a href=${this.step.url} target="_blank" rel="opener">
+            <mwc-button raised>
+              ${this.localize("ui.panel.page-authorize.external.open_site")}
+            </mwc-button>
+          </a>
+        </div>
+      </div>
+    `;
+  }
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+    window.open(this.step.url);
+    window.addEventListener("message", async (message: MessageEvent) => {
+      if (message.data.type === "externalCallback") {
```

Comment:
Can we do a check for `origin` and `source` of the message?

Reply:
Yes, sounds good.  I have this working for the window.open (following what the previous PR did). How do you track the source of the button that opens the window above?

Reply:
I dont think there is a way with just an anchor link. You could make it a click event and open the window programmatically with `window.open`. Hoping that will not cause issues with popup blockers.

Reply:
OK -- I changed this to use a button to run the same code thats run when first loading. Perhaps that is reasonable...